docs/proposals/operator-custom-crd-download-strategy/README.md

```diff
@@ -0,0 +1,110 @@
+is identical to that of the operator's release version, which may not always be true. When the version of the Karmada control plane components is not specified, the default behavior is to assume
+those components should have the same version as the operator's release version. In that scenario, the assumption is justified and holds true. However, when creating a new Karmada instance, it's possible to explicitly
+specify the version of the control plane components. In that scenario, this assumption breaks, and we encounter the risk of the operator installing the wrong CRDs for that Karmada instance.
+
+The default behavior is sensible and will be retained. However, it should not be the behavior when the intent is to create a new Karmada instance with control plane components of a different version than that of the operator's release version. 
+To support that scenario, we require the ability to explicitly specify, as part of the Karmada resource, how the operator should download the CRDs tarball for that Karmada instance.
+
+In addition to making sure required CRDs are downloaded, the Karmada operator also employs a caching strategy to avoid re-downloading CRDs it has already downloaded. It's a cache on the local file system.
+Downloaded CRDs are stored at a well-known location on the local file system based the release version of the Karmada operator. When provisioning a new Karmada instance, before attempting to download the required CRDs, the operator will fist check if
+they already exist in the cache. To support custom downloading strategies, we require this caching strategy be extended to be more generic.
+
+## Design Details
+
+To support custom CRD download strategies, we need to add the new `CRDTarball` field to the Karmada resource spec:
+```go
+// CRDDownloadPolicy specifies a policy for how the operator will download the Karmada CRD tarball
+type CRDDownloadPolicy string
+
+const (
+	// DownloadAlways instructs the Karmada operator to always download the CRD tarball from a remote location.
+	DownloadAlways CRDDownloadPolicy = "Always"
+
+	// DownloadIfNotPresent instructs the Karmada operator to download the CRDs tarball from a remote location only if it is not yet present in the local cache.
+	DownloadIfNotPresent CRDDownloadPolicy = "IfNotPresent"
+)
+
+// HTTPSource specifies how to download the CRD tarball via either HTTP or HTTPS protocol.
+type HTTPSource struct {
+	// URL specifies the URL of the CRD tarball resource.
+	// +optional
+	URL *string `json:"url,omitempty"`
+}
+
+// CRDTarball specifies the source from which the Karmada CRD tarball should be downloaded, along with the download policy to use.
+type CRDTarball struct {
+	// HTTPSource specifies how to download the CRD tarball via either HTTP or HTTPS protocol.
+	// +optional
+	HTTPSource *HTTPSource `json:"httpSource,omitempty"`
```

Comment:
Looks great!

Would you mind [Squash commits](https://karmada.io/docs/next/contributor/github-workflow/#squash-commits) ? and we are ready to move forward if no further comments from other guys, like @chaunceyjiang .

Reply:
Perfect! Just squashed. Thanks again for the constructive feedback.